ci/ci/ci.py

```diff
@@ -213,20 +213,39 @@ def filter_wbs(wbs, pred):
     return [{**wb, 'prs': [pr for pr in wb['prs'] if pred(pr)]} for wb in wbs]
 
 
+def is_pr_author(gh_username, pr_config):
+    return gh_username == pr_config['author']
+
+
+def is_pr_reviewer(gh_username, pr_config):
+    return gh_username in pr_config['assignees']
+
+
+def pr_requires_action(gh_username, pr_config):
+    build_state = pr_config['build_state']
+    review_state = pr_config['review_state']
+    return (
+        is_pr_author(gh_username, pr_config) and (build_state == 'failure' or review_state == 'changes_requested')
+    ) or (is_pr_reviewer(gh_username, pr_config) and review_state == 'pending')
```

Comment:
this would look prettier in lisp..

Reply:
I do quite like function notation for and and or.